cli/pkg/backend/aws/k8s_compute_backend.go

```diff
@@ -203,38 +206,128 @@ func (k8s *K8SComputeBackend) PrintLogs(ctx context.Context, stackName string, s
-			return errors.Wrapf(err, "unable to retrieve logs from pod %s", pod.Name)
+			logrus.Error(err.Error())
 		}
-		defer logs.Close()
-		reader := bufio.NewScanner(logs)
+	}
+	return nil
+}
 
-		for reader.Scan() {
-			logrus.Info(string(reader.Bytes()))
-		}
-		logrus.Infof("... done streaming ...")
+func (k8s *K8SComputeBackend) streamPodLogs(ctx context.Context, pod corev1.Pod, follow bool) error {
+	logrus.Infof("... streaming logs from pod %s ...", pod.Name)
+
+	logs, err := k8s.ClientSet.CoreV1().Pods(k8s.HappyConfig.K8SConfig().Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+		Follow: follow,
+	}).Stream(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "unable to retrieve logs from pod %s", pod.Name)
 	}
+	defer logs.Close()
+	reader := bufio.NewScanner(logs)
+
+	for reader.Scan() {
+		logrus.Info(string(reader.Bytes()))
+	}
+	logrus.Infof("... done streaming ...")
 	return nil
 }
 
 func (k8s *K8SComputeBackend) RunTask(ctx context.Context, taskDefArn string, launchType config.LaunchType) error {
-	// TODO: not implemented
-	return errors.New("not implemented")
+	// Get the cronjob and create a job out of it
+	if len(taskDefArn) >= 2 {
+		if taskDefArn[0] == '"' && taskDefArn[len(taskDefArn)-1] == '"' {
+			taskDefArn = taskDefArn[1 : len(taskDefArn)-1]
+		}
+	}
+	cronJob, err := k8s.ClientSet.BatchV1().CronJobs(k8s.HappyConfig.K8SConfig().Namespace).Get(ctx, taskDefArn, v1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve a template cronjob")
+	}
+
+	jobId := fmt.Sprintf("%s-%s-job", taskDefArn, uuid.NewUUID())
+	jobDef := k8s.createJobFromCronJob(cronJob, jobId)
+	jb, err := k8s.ClientSet.BatchV1().Jobs(k8s.HappyConfig.K8SConfig().Namespace).Create(ctx, jobDef, v1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to create a k8s job")
+	}
+
+	logrus.Info("Waiting for all the pods to start")
+	// Make sure pods have started
+	counter := 0
+	var pods *corev1.PodList
+	for {
+		time.Sleep(5 * time.Second)
+		pods, err = k8s.getJobPods(ctx, jb.Name)
+		if err != nil {
+			logrus.Errorf("unable to retrieve job pods, will retry")
+			continue
+		}
+
+		if len(pods.Items) == 0 {
+			logrus.Errorf("unable to find any pods created by the job, will retry")
+			continue
+		}
+
+		if counter > 20 {
+			return errors.New("Unable to find any running pod spawned by the task")
+		}
+
+		allPodsReady := true
+		for _, pod := range pods.Items {
+			if pod.Status.Phase == corev1.PodPending {
+				allPodsReady = false
+				break
+			}
+		}
+		if !allPodsReady {
+			for _, pod := range pods.Items {
+				if pod.Status.Phase == corev1.PodPending {
+					allPodsReady = false
+					break
+				}
+				logrus.Infof("%s: %s, %s", pod.Name, pod.Status.Phase, pod.Status.Message)
+			}
+			continue
+		}
+
+		break
+	}
+
+	logrus.Infof("Found %d pods", len(pods.Items))
+	for _, pod := range pods.Items {
+		err = k8s.streamPodLogs(ctx, pod, true)
+		if err != nil {
+			logrus.Error(err.Error())
+		}
+	}
```

Comment:
TODO for later, but we should probably interleave these instead of appending them to one another